tests/bsim/bluetooth/mesh/src/test_proxy_sol.c

```diff
@@ -0,0 +1,293 @@
+	BT_MESH_MODEL_OD_PRIV_PROXY_SRV,
+};
+
+static struct bt_mesh_elem elems[] = {
+	BT_MESH_ELEM(0, models, BT_MESH_MODEL_NONE),
+};
+
+static const struct bt_mesh_comp comp = {
+	.cid = TEST_VND_COMPANY_ID,
+	.vid = 0xaaaa,
+	.pid = 0xbbbb,
+	.elem = elems,
+	.elem_count = ARRAY_SIZE(elems),
+};
+
+bool isTesterAddress(void)
+{
+	return bt_mesh_primary_addr() == tester_cfg.addr;
+}
+
+static uint8_t proxy_adv_type_get(uint8_t adv_type, struct net_buf_simple *buf)
+{
+	uint8_t type;
+	uint8_t len = buf->len;
+
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_IND || len < 12) {
+		return 0xFF;
+	}
+
+	(void)net_buf_simple_pull_mem(buf, 11);
+	type = net_buf_simple_pull_u8(buf);
+
+	if (len != ((type == BEACON_TYPE_NET_ID) ? BEACON_NET_ID_LEN : OTHER_ADV_TYPES_LEN)) {
+		return 0xFF;
+	}
+
+	return type;
+}
+
+static void scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_simple *ad)
+{
+	uint8_t type = proxy_adv_type_get(info->adv_type, ad);
+
+	if (isTesterAddress() && type == BEACON_TYPE_PRIVATE_NET_ID) {
+		k_sem_give(&beacon_sem);
+	}
+}
+
+static struct bt_le_scan_cb scan_cb = {
+	.recv = scan_recv,
+};
+
+static void tester_configure(void)
+{
+	uint8_t status;
+
+	k_sem_init(&beacon_sem, 0, 1);
+	bt_le_scan_cb_register(&scan_cb);
+
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &comp);
+
+	ASSERT_OK(bt_mesh_provision(test_net_key, 0, 0, 0, tester_cfg.addr, tester_cfg.dev_key));
+	ASSERT_OK(bt_mesh_cfg_cli_app_key_add(0, tester_cfg.addr, 0, 0, test_app_key, &status));
+}
+
+static void iut_configure(void)
+{
+	uint8_t status;
+
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &comp);
+
+	/* Network stored in flash during immediate_replay_attack */
+	if (!bt_mesh_is_provisioned()) {
+		ASSERT_OK(bt_mesh_provision(test_net_key, 0, 0, 0, iut_cfg.addr, iut_cfg.dev_key));
+	}
+
+	ASSERT_OK(bt_mesh_cfg_cli_app_key_add(0, iut_cfg.addr, 0, 0, test_app_key, &status));
+	ASSERT_OK(bt_mesh_cfg_cli_gatt_proxy_set(0, iut_cfg.addr, BT_MESH_GATT_PROXY_DISABLED,
+						 &status));
+	ASSERT_OK(bt_mesh_priv_beacon_cli_gatt_proxy_set(0, iut_cfg.addr,
+							 BT_MESH_GATT_PROXY_DISABLED, &status));
+	ASSERT_OK(bt_mesh_od_priv_proxy_set(1));
+}
+
+static int sol_fixed_pdu_create(struct bt_mesh_subnet *sub, struct net_buf_simple *pdu)
+{
+	static uint32_t fixed_seq_n = 0x123456;
+
+	net_buf_simple_add_u8(pdu, sub->keys[SUBNET_KEY_TX_IDX(sub)].msg.nid);
+	net_buf_simple_add_u8(pdu, 0x80);
+	net_buf_simple_add_le24(pdu, sys_cpu_to_be24(fixed_seq_n));
+	net_buf_simple_add_le16(pdu, sys_cpu_to_be16(bt_mesh_primary_addr()));
+	net_buf_simple_add_le16(pdu, 0x0000);
+
+	ASSERT_OK(bt_mesh_net_encrypt(&sub->keys[SUBNET_KEY_TX_IDX(sub)].msg.enc, pdu, 0,
+				      BT_MESH_NONCE_SOLICITATION));
+
+	ASSERT_OK(bt_mesh_net_obfuscate(pdu->data, 0,
+					&sub->keys[SUBNET_KEY_TX_IDX(sub)].msg.privacy));
+
+	net_buf_simple_push_u8(pdu, 0);
+	net_buf_simple_push_le16(pdu, BT_UUID_MESH_PROXY_SOLICITATION_VAL);
+
+	return 0;
```

Comment:
Do you need to return anything at all if it is always zero?
Here and for the others below.